projects/ADuCM3029_demo_cn0428_cn0429/src/cli.c

```diff
@@ -0,0 +1,986 @@
+	UART_TX("                                  <site> = 1, 2, 3, 4"_EOS);
+	UART_TX("sensorsconnected               -Print which sensors are connected"_EOS);
+	UART_TX("readconfigs                    -Read sensor configurations"_EOS);
+	UART_TX("readtemp                       -Read temperature from on-board sensor"_EOS);
+	UART_TX("readhum                        -Read humidity from on-board sensor"_EOS);
+	UART_TX("setmeastime <time>             -Set ADC sampling time in msec, default - 500"_EOS);
+	UART_TX("                                  <time> = <50-32000>"_EOS);
+	UART_TX("                                  ADC performs avg of 10 samples at this interval"_EOS);
+	UART_TX("startmeas                      -Start ADC sampling at configured interval"_EOS);
+	UART_TX("stopmeas                       -Stop ADC sampling the sensor"_EOS);
+	UART_TX("setrtia <rtia>                 -Configure TIA resistance"_EOS);
+	UART_TX("                                  <rtia> = <0-26> equals 0R - 512k"_EOS);
+	UART_TX("                                  see user guide for detailed table"_EOS);
+	UART_TX("setrload <rload>               -Configure sensor load resistor"_EOS);
+	UART_TX("                                  <rload> = <0-7> equals 0R - 3k6"_EOS);
+	UART_TX("                                  see user guide for detailed table"_EOS);
+	UART_TX("setvbias <vbias>               -Configure sensor bias voltage"_EOS);
+	UART_TX("                                  <vbias> = bias voltage in mV"_EOS);
+	UART_TX("setsensitivity <sens>          -Configure sensitivity"_EOS);
+	UART_TX("                                  <sens> = sensitivity in nA/ppm"_EOS);
+	UART_TX("runeis                         -Run impedance spectroscopy"_EOS);
+	UART_TX("readeis                        -Read impedance spectroscopy results"_EOS);
+	UART_TX("readeisfull                    -Read impedance spectroscopy FULL results"_EOS);
+	UART_TX("readrcal                       -Read 200R calibration resistor"_EOS);
+	UART_TX("runpulse                       -Run pulse test. Set amplitude and duration first!"_EOS);
+	UART_TX("readpulse                      -Read pulse test results"_EOS);
+	UART_TX("pulseamplitude <ampl>          -Set amplitude for pulse test"_EOS);
+	UART_TX("                                  <ampl> = amplitude in mV, 1 - 3 mV"_EOS);
+	UART_TX("pulseduration <dur>            -Set duration for pulse test"_EOS);
+	UART_TX("                                  <dur> = duration in ms, 10 to 200"_EOS);
+	UART_TX("readsensors                    -Read the ppb value of all sensors"_EOS);
+	UART_TX("setupdaterate <time>           -Configure update rate for stream mode"_EOS);
+	UART_TX("stopread                       -Stop sensor data reading"_EOS);
+	UART_TX("\r\n"_EOS);
+}
+
+/**
+   @brief Finds the next command line argument
+
+   @param args - pointer to the arguments on the command line.
+
+   @return pointer to the next argument.
+
+**/
+uint8_t *FindArgv(uint8_t *args)
```

Comment:
this looks un-needed; a main() function always has an `argv` and an `argc`, so that you don't need to do string lookups byte-by-byte;

Reply:
These are space-separated arguments that come in through the uart at run-time, not program arguments that would come in through the main at the start of execution. See the aducm3029 cli project and the cn0326 project that implemented this previously.